src/components/AnnotationProps.tsx

```diff
@@ -142,5 +142,82 @@ export default interface AnnotationProps {
+   */
+  calloutOffsetX?: number;
+
+  /**
+   * An Y offset that changes the annotation callout’s default placement.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/mapkit/annotation/2973821-calloutoffset}
+   */
+  calloutOffsetY?: number;
+
+  /**
+   * A Boolean value that determines whether the map shows an annotation’s callout.
+   * If the title is empty, the framework can’t show the standard callout even if property is true.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/mapkit/annotation/2973820-calloutenabled}
+   */
+  calloutEnabled?: boolean;
+
+  /**
+   * Returns an element to use as a custom accessory on the left side of the callout content area.
+   *
+   * When MapKit JS creates a callout for a selected annotation and the annotation’s callout
+   * delegate has no calloutElementForAnnotation method, the framework calls
+   * calloutContentForAnnotation method instead — if it’s defined — on the delegate with the
+   * annotation as a parameter.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/annotationcalloutdelegate/2991150-calloutleftaccessoryforannotatio}
+   */
+  calloutLeftAccessoryForAnnotation?: JSX.Element;
+
+  /**
+   * Returns an element to use as a custom accessory on the right side of the callout content area.
+   *
+   * When MapKit JS creates a callout for a selected annotation and the annotation’s callout
+   * delegate has no calloutElementForAnnotation method, the framework calls
+   * calloutContentForAnnotation method instead — if it’s defined — on the delegate with the
+   * annotation as a parameter.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/annotationcalloutdelegate/2991151-calloutrightaccessoryforannotati}
+   */
+  calloutRightAccessoryForAnnotation?: JSX.Element;
+
+  /**
+   * Returns custom content for the callout bubble.
+   *
+   * When MapKit JS creates a callout for a selected annotation and the annotation’s callout
+   * delegate has no calloutElementForAnnotation method, the framework calls
+   * calloutContentForAnnotation method instead — if it’s defined — on the delegate with the
+   * annotation as a parameter.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/annotationcalloutdelegate/2991148-calloutcontentforannotation}
+   */
+  calloutContentForAnnotation?: JSX.Element;
+
+  /**
+   * Returns an element representing a custom callout.
+   *
+   * When MapKit JS creates a callout for a selected annotation and the annotation’s callout
+   * delegate has no calloutElementForAnnotation method, the framework calls
+   * calloutContentForAnnotation method instead — if it’s defined — on the delegate with the
+   * annotation as a parameter.
+   * @see {@link https://developer.apple.com/documentation/mapkitjs/annotationcalloutdelegate/2991148-calloutcontentforannotation}
+   */
+  calloutElementForAnnotation?: JSX.Element;
```

Comment:
I agree, though this is the same behavior as in mapkitJS. Generally speaking the Typescript solution preventing the setting of both of them would be the the cleanest I guess, I don't think we should implement an error, which mapkitJS wouldn't throw on it's own...

Reply:
I tried to implement a conditional type, however, I don't really know how to do that with interfaces.

I was trying something like:

```
...Markerprops extends CalloutProps ...

//doesn't work: Cannot mix types and interfaces in that case
interface CalloutProps extends CalloutElemProps {
...calloutEnabled ...
}


type CalloutElemProps = CalloutElementProps | CalloutContentProps;

interface CalloutElementProps {
...calloutElement...
}

interface CalloutContentProps {
...calloutContent...
...calloutLeft...
...calloutRight...
}
```

Reply:
I thought there would be a way to fix this by using `type` instead of `interface`. Something like:

```ts
export type MarkerProps = {
  // ...
  calloutEnabled?: boolean;
} & ({
  calloutLeftAccessory?: ReactNode;
  calloutRightAccessory?: ReactNode;
  calloutContent?: ReactNode;

  calloutElement?: never;
} | {
  calloutLeftAccessory?: never;
  calloutRightAccessory?: never;
  calloutContent?: never;

  calloutElement?: ReactNode;
});
```

But I’ve just tried it and while it produces the correct type (e.g. it only allows to set either calloutElement or the other properties at once), it hurts the developer experience (VS Code isn’t capable of showing the attribute documentation when using a union type for the props). So I think that it's best to keep using your current implementation.

Reply:
Ok, should we still implement a error which is thrown if both attributes are set or is it rather not necessary?

Reply:
No, I don’t think it is necessary since MapKit JS doesn’t do it.